### ISSUE TITLE
`unused_local_variable` false positive when used with switch expression

### ISSUE THREAD

**Issue by Pante** [dart-lang/linter#4623]
**Describe the issue**
There's currently a false positive when using cascading functions with a switch expression.

**To Reproduce**
The following code snippet:
```dart
void a() {
  final foo = switch (true) {
    _ => Foo(),
  }..function()..function();
}

class Foo {
  void function() {}
}
```

Produces: `The value of the local variable 'foo' isn't used.` However, removing `foo` causes a compile error.
